[es] Translate docs/guidance/reference-implementations/mastodon/index.md (#10883)
open-telemetry/opentelemetry.io: content/es/docs/guidance/reference-implementations/mastodon/index.md

---
title:
  'Mastodon: ejecución de Collectors de OpenTelemetry en producción con un
  equipo pequeño'
linkTitle: Mastodon
default_lang_commit: 11753c0e99bbc1b62606d4c819736c777dfb0e98

> cSpell:ignore: Fediverse otelbin Sidekiq Sloughter Öjeling

Reviewer: ```suggestion
cSpell:ignore: otelbin Sidekiq Sloughter Öjeling
```

---

Por [Juliano Costa](https://github.com/julianocosta89) (Datadog),
[Tristan Sloughter](https://github.com/tsloughter) (community),
[Johanna Öjeling](https://github.com/johannaojeling) (Grafana Labs),
[Damien Mathieu](https://github.com/dmathieu) (Elastic),
[Tim Campbell](https://github.com/timetinytim) (Mastodon) | 18 de marzo de 2026

Esta implementación de referencia describe cómo Mastodon, una organización sin
ánimo de lucro que opera a escala global con un equipo notablemente pequeño,
ejecuta el OpenTelemetry Collector en producción.

## Mastodon de un vistazo {#mastodon-at-a-glance}

[Mastodon](https://joinmastodon.org) es una plataforma de redes sociales
descentralizada, gratuita y de código abierto, operada por una organización sin
ánimo de lucro.

La descentralización no es aquí un término de marketing, sino un principio
arquitectónico central. Cualquiera puede
[ejecutar su propio servidor de Mastodon](https://docs.joinmastodon.org/user/run-your-own/),
y esos servidores operados de forma independiente interoperan mediante

> protocolos abiertos como parte de lo que se conoce como el _Fediverse_: una red

Reviewer: ```suggestion
protocolos abiertos como parte de lo que se conoce como el fediverso: una red
```
This is a well-known word in Spanish, so I would translate it. Most words we use in English, but this once I always hear the translation.

federada de plataformas sociales independientes que se comunican entre sí usando
protocolos estandarizados como ActivityPub. Al igual que ocurre con el correo
electrónico, los usuarios pueden comunicarse entre instancias independientemente
de quién las opere.

> Esta filosofía condiciona no solo las decisiones de funcionalidades de Mastodon,

Reviewer: ```suggestion
Esta filosofía no solo condiciona las decisiones de producto de Mastodon,
```
Sounds a bit more natural, but the original is also good; it just sounds a bit more like a translation from English.

ejecuten, el equipo quería una solución de telemetría que preservara la libertad
del operador.

OpenTelemetry se convirtió en la opción predeterminada porque permite que cada
operador de un servidor Mastodon decida cómo —o si— se recopila la telemetría.

Mediante una sencilla
[configuración por variables de entorno](https://docs.joinmastodon.org/admin/config/#otel),
los operadores pueden elegir:

- Enviar la telemetría directamente a un backend de observabilidad (usando
  únicamente la configuración del SDK de Ruby)
- Enrutar la telemetría a través de un OpenTelemetry Collector
- Deshabilitar la telemetría por completo

La organización central de Mastodon no realiza seguimiento de cómo gestionan la
observabilidad las instancias externas. Lo que importa es que la telemetría
emitida se ajuste estrictamente a las
**[convenciones semánticas de OpenTelemetry](/docs/specs/semconv/)**, lo que la
hace utilizable en cualquier lugar.

Este enfoque evita los modelos de datos específicos de proveedor y garantiza la
compatibilidad con el ecosistema más amplio de OpenTelemetry, sin que Mastodon
tenga que mantener sus propias convenciones.

> ## Arquitectura del Collector: uno por namespace, y no más {#collector-architecture-one-per-namespace-no-more}

Reviewer: ```suggestion
## Arquitectura del Collector: uno solo por namespace {#collector-architecture-one-per-namespace-no-more}
```

Para mantener la sobrecarga operativa lo más baja posible, Mastodon se apoya en:

- El [OpenTelemetry Operator](/docs/platforms/kubernetes/operator/) para
  Kubernetes
- Argo CD para despliegues y promociones basados en Git

Cada Collector se define como un recurso personalizado `OpenTelemetryCollector`.
A partir de ahí, Kubernetes se encarga automáticamente de la reconciliación, los
reinicios y la gestión del ciclo de vida.

> «Básicamente, solo necesitamos crear un archivo yaml para cada objeto
> `OpenTelemetryCollector` que necesitemos crear, y Argo se encarga de
> desplegar/actualizar automáticamente lo que necesitamos.»

Este modelo proporciona:

- Configuración declarativa
- Recuperación automática ante fallos
- Auditabilidad clara a través del historial de Git

Cabe destacar que Mastodon no impone límites estrictos de CPU o memoria a los
pods del Collector. En la práctica, el consumo de recursos se ha mantenido
insignificante en comparación con el resto de la plataforma.

## Gestión del tráfico mediante el muestreo {#traffic-management-through-sampling}

En lugar de basarse en límites de recursos, Mastodon controla la sobrecarga de

> observabilidad principalmente mediante el muestreo basado en la cola (tail-based

Reviewer: ```suggestion
observabilidad principalmente mediante el muestreo basado en cola (tail-based
```
Or simply leave _tail-based sampling_, we usually talk about tail sampling even in Spanish.

sampling).

- En mastodon.social, las trazas exitosas se muestrean aproximadamente al 0,1 %,
  lo que da como resultado solo unas pocas docenas de trazas por minuto a pesar
  del tráfico extremadamente alto.
- En mastodon.online, el muestreo es ligeramente más permisivo, pero sigue los
  mismos principios.
- Todas las trazas de error se recopilan siempre, lo que garantiza una
  visibilidad completa de los fallos.

Este enfoque mantiene el volumen de datos predecible al tiempo que conserva los
datos de diagnóstico de alto valor.

## Configuración: con criterio propio, pero mínima {#configuration-opinionated-but-minimal}

Mastodon usa la distribución OpenTelemetry Collector Contrib, principalmente por
conveniencia: incluye todo lo que necesitan sin requerir compilaciones
personalizadas.

La configuración se centra en:

> - Ingesta de OTLP para todas las señales

Reviewer: ```suggestion
- Ingesta OTLP para todas las señales
```

    - context: span
      conditions:
        - attributes["code.namespace"] != nil
      statements:
        - set(attributes["resource.name"], Concat([attributes["code.namespace"],
          attributes["code.function"]], "#"))
```

En la regla del procesador transform anterior, han configurado una condición
para establecer `resource.name` (un atributo específico de Datadog) con el valor
de `code.namespace#code.function`. Con esa configuración, cada vez que el span
llegaba al backend, podía asignarse al nombre que habían definido. A pesar de
esa curva de aprendizaje, la experiencia general ha superado las expectativas.

> «Básicamente puedes hacer lo que quieras. Superó mis expectativas. Todo
> funciona bastante bien.»

Esa fiabilidad y flexibilidad son las razones por las que Mastodon sigue usando
el OpenTelemetry Collector en producción.

## Consejos para equipos pequeños {#advice-for-small-teams}

A partir de la experiencia de Mastodon, destacan algunas lecciones:

- **Mantén la arquitectura simple**: un único Collector puede llegar muy lejos
- **Confía en los operadores de Kubernetes** para la gestión del ciclo de vida
- **Usa el muestreo** para controlar los costes

> - **Cíñete a las convenciones semánticas** para evitar el bloqueo (lock-in) a

Reviewer: ```suggestion
- **Cíñete a las convenciones semánticas** para evitar la dependencia de proveedor (lock-in) a
```
